analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -95,8 +97,39 @@ def merge_ctu_func_maps(ctu_dir, ctu_func_map_file, ctu_temp_fnmap_folder):
             shutil.rmtree(fnmap_dir, ignore_errors=True)
 
 
+def generate_invocation_list(triple_arch, action, source, config, env):
+    """ Generates the invocation for the source file of the current
+    compilation command. Used during on-demand analysis. The invocation list
+    is a mapping from absolute paths of the source files to the parts of the
+    compilation commands that are used to generate from each one of them. """
+
+    triple_arch_dir = Path(config.ctu_dir, triple_arch)
+    # Ensure, that architecture directory exists.
+    triple_arch_dir.mkdir(parents=True, exist_ok=True)
+
+    invocation_list = triple_arch_dir / 'invocation-list.yml'
+
+    source_path = Path(source).resolve()
+
+    cmd = ctu_triple_arch.get_compile_command(action, config, source)
+    # __clang__analyzer__ macro needs to be set in the imported TUs too.
+    cmd.extend(['-D__clang_analyzer__', '-w'])
+
+    # Dump the mapping as a JSON dictionary, then strip the braces.
+    # The resulting line is a proper YAML mapping entry.
+    # TODO: Use proper YAML serialization
```

Comment:
the yaml library will be introduced in another PR too so I think we can use that to generate proper yaml files.

Reply:
yeeey, cant wait for the shiny new YAML library :D ! 👍